File: leetcode-python-go/1233. 删除子文件夹/main.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

func removeSubfolders(folder []string) (ans []string) {
	sort.Strings(folder)
	ans = append(ans, folder[0])
	for _, f := range folder[1:] {
		last := ans[len(ans)-1]
		if !strings.HasPrefix(f, last) || f[len(last)] != '/' {
			ans = append(ans, f)
		}
	}
	return
}

func main() {
	folder := []string{"/a/b/c", "/a/b/ca", "/a/b/d"}
	results := removeSubfolders(folder)
	fmt.Println(results)
}

File: leetcode-python-go/1233. 删除子文件夹/main.py
from typing import List

folder = ["/a","/a/b","/c/d","/c/d/e","/c/f"]

def removeSubfolders(folder: List[str]) -> List[str]:
    folder.sort()
    ans = [folder[0]]
    for i in range(1, len(folder)):
        if not ((pre := len(ans[-1])) < len(folder[i]) and ans[-1] == folder[i][:pre] and folder[i][pre] == "/"):
            ans.append(folder[i])
    return ans

print(removeSubfolders(folder))